### PR TITLE
Perf tuning changes

### PR DESCRIPTION
**Pull request by snambakam**
* For aarch64 -
  - Remove CONFIG_NF_CONNTRACK_PROCFS
  - Remove CONFIG_TRACE_IRQFLAGS
  - Remove CONFIG_TRACE_IRQFLAGS_NMI
  - Remove CONFIG_IRQSOFF_TRACER
  - Remove CONFIG_PREEMPTIRQ_TRACEPOINTS
* For both x86_64 & aarch64
  - Remove CONFIG_TRANSPARENT_HUGEPAGE_ALWAYS
  - Add CONFIG_TRANSPARENT_HUGEPAGE_MADVISE

###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] If you are adding/removing a .spec file that has multiple-versions supported, please add [@microsoft/cbl-mariner-multi-package-reviewers](https://github.com/orgs/microsoft/teams/cbl-mariner-multi-package-reviewers) team as reviewer [(Eg. golang has 2 versions 1.18, 1.21+)](https://github.com/microsoft/azurelinux/tree/2.0/SPECS/golang)
- [x] Ready to merge

---

###### Summary

Performance tuning changes.

###### Change Log 

* For aarch64 -
  - Remove CONFIG_NF_CONNTRACK_PROCFS
  - Remove CONFIG_TRACE_IRQFLAGS
  - Remove CONFIG_TRACE_IRQFLAGS_NMI
  - Remove CONFIG_IRQSOFF_TRACER
  - Remove CONFIG_PREEMPTIRQ_TRACEPOINTS
* For both x86_64 & aarch64
  - Remove CONFIG_TRANSPARENT_HUGEPAGE_ALWAYS
  - Add CONFIG_TRANSPARENT_HUGEPAGE_MADVISE

###### Does this affect the toolchain? 
YES due to kernel-headers version change.

###### Associated issues 
N/A

###### Links to CVEs  
N/A

###### Test Methodology
- Pipeline build id 559938

### CODE REVIEW

File: SPECS/kernel-headers/kernel-headers.spec [hunk 1]
```diff
@@ -75,6 +75,9 @@ done
 %endif
 
 %changelog
+* Apr 29 2024 Sriram Nambakam <snambakam@microsoft.com> - 6.6.22.1-3
```
Comment (anphel31) [inline SPECS/kernel-headers/kernel-headers.spec:78]: Can you add the day ("Mon")

**Review by anphel31 — COMMENTED** [review]
Can you add a link to a buddybuild?

File: SPECS/kernel-headers/kernel-headers.spec (hunk 1 above)
Comment (snambakam) [reply SPECS/kernel-headers/kernel-headers.spec]: Fixed

**Review by anphel31 — APPROVED** [review]

**Comment by anphel31** [comment]
I ran a buddybuild to confirm the (unsigned) kernel build for both architectures: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=560765&view=results

**Review by rlmenge — APPROVED** [review]

**Pull request merged**